src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,97 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
+import { pattern } from '../utils/patterns';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const validationUrl = (string: string): boolean => {
+  return pattern.test(string);
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isImgUrlValid, setIsImgUrlValid] = useState(false);
+  const [isImdbIdValid, setIsImdbIdValid] = useState(false);
+
+  const isAllChackedFieldsValid = Boolean(!isImgUrlValid && !isImdbIdValid);
+  const isAllowSubmit = Boolean(title && imgUrl && imdbUrl && imdbId
+    && isAllChackedFieldsValid);
+
+  const addMovie = () => {
+    const newMovie: Movie = {
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    };
+
+    setCount(current => current + 1);
+
+    onAdd(newMovie);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={addMovie}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        onChange={(str) => setTitle(str)}
```

Comment:
```suggestion
        onChange={setTitle}
```
It can be simplified

Reply:
Thanks.